avahi-daemon/avahi-daemon.service.in

```diff
@@ -26,6 +26,24 @@ ExecStart=@sbindir@/avahi-daemon -s
 ExecReload=@sbindir@/avahi-daemon -r
 NotifyAccess=main
 
+ProtectSystem=strict
+ReadWritePaths=@avahi_runtime_dir@/avahi-daemon/
```

Comment:
I think one could consider to use `RuntimeDirectory=avahi-daemon` here instead (and thus can also drop the extra Makefile mangling snippet). See https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RuntimeDirectory=

AFAIR using ReadWritePaths= will fail the unit when the directory does not exist (and who creates /run/avahi-daemon before the unit starts?).

Reply:
Thanks.  Actually, I think /run/avahi-daemon/ ends up being created by avahi-daemon.socket.

I think `RuntimeDirectory=` is "recommended", and it's been around since systemd-211 (2014-03-12). So it sounds like the way to go. I *think* the interaction with the socket unit should work well.

It just looks a bit funny, so I want to take a bit longer to think about it. Maybe I'll add some commentary in the commit message.